src/hashtable.py

```diff
@@ -51,8 +51,16 @@ def insert(self, key, value):
 
         Fill this in.
         '''
-        pass
-
+        index = self._hash_mod(key)
+        node = self.storage[index]
+        while node is not None and key is not node.key:
+            if node.next is not None:
+                node = node.next
+            else:
+                node.next = LinkedPair(key, value)
+                return
+        self.storage[index] = LinkedPair(key, value)
+        return
 
```

Comment:
I don't feel like this is handling overwriting correctly.  It looks like you're overwriting the whole linked list when a key matches, instead of changing the value in the list node itself